## store infromation needed for all protocols
locations:
  bedroom:
    x: -1.08
    y: -1.4811692237854004
    yaw: -1.5708   # -90 degrees

  living_room:
    x: 1.0967974662780762
    y: 1.9037660360336304
    yaw: 0.0       # facing forward
  home:
    x: 2.6955928802490234 #0.85 #
    y: 3.3216073513031006 #1.014 #
    yaw: -1.5708   # -90 degrees

  dining:
    x: 5.066422939300537
    y: 3.4881248474121094
    yaw: -1.5708   # -90 degrees



## need to be integerated to have robot interface get its info 
person_location_topic: "person_location"
robot_location_topic: "robot_location"

# available type: ## todo set it in a way to verifiy htat yaml doesnt include 
## undefined types: text, audio, video, question_answer 
protocols:
  TwoReminderProtocol:
    medicine_am:
      permissible_locations: TO BE INCORPORATED
      low_level:
        first_reminder: "/home/hello-robot/smarthome_ws/src/smart_home_robot/smart_home_pytree/audio_resources/food_reminder_dain.mp3" #"please take your morning medicine"
        type_first: audio
        second_reminder: "/home/hello-robot/smarthome_ws/src/smart_home_robot/smart_home_pytree/audio_resources/food_reminder.mp3" #"This is the second reminder to take your morning medicine"
        type_second: audio
        wait_time_between_reminders: 100
      high_level:
        priority: 2
        reset_pattern:
          type: periodic
          minutes: 60
        requirements: 
          time: {'from': '10:00', 'to': '11:00'}

    medicine_pm:
      low_level:
        first_reminder: "please take your night medicine"
        type_first: text
        second_reminder: "This is the second reminder to take your night medicine"
        type_second: text
        wait_time_between_reminders: 5
      high_level: 
        priority: 2
        requirements: 
          day: [monday, wednesday, friday]
          time: {'from': '22:00', 'to': '23:00'}

## add topics with /
  CoffeeProtocol:
    coffee:
      low_level:
        first_text: "please clean the dishes"
      high_level:
        priority: 3
        reset_pattern:
          type: periodic
          hours: 1
          minutes: 30
        requirements: 
          event: 
            - {'topic': '/coffee', 'value': True}
            - {'topic': '/coffee_pot', 'value': True}
          time: {'from': '09:00', 'to': '13:00'}

  MoveAwayProtocol:
    move_away:
      low_level:
        position_state_key: position # indicates which key it should refer to in the state dictionary
      high_level:
        priority: 1
        requirements: 
          event: 
            - {'topic': '/move_away', 'value': True}  